src/basic/fsverity-util.c

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#if HAVE_LINUX_FSVERITY_H
+
+#include <linux/fsverity.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include "alloc-util.h"
+#include "errno-util.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "fsverity-util.h"
+#include "string-util.h"
+
+int fsverity_enable(int fd, const char *path, const void *signature, size_t signature_size) {
+        struct fsverity_enable_arg params = {
+                .hash_algorithm = FS_VERITY_HASH_ALG_SHA256,
+                .sig_ptr = (uintptr_t) signature,
+                .sig_size = signature_size,
+                /* Up until kernel 6.3, the fsverity block size must be equal to the page size (and FS block size) */
+                .block_size = page_size(),
+                .version = 1,
+        };
+        _cleanup_close_ int read_only_fd = -EBADF;
+        int r;
+
+        assert(fd >= 0 || path);
+        assert(path);
+        assert(signature || signature_size == 0);
+
+        if (fd == -EBADF) {
```

Comment:
we almost always accept any < 0 value for fd as "not specified", why not here? i.e. change to → `if (fd < 0)`